### ISSUE TITLE
Sticking stays disabled when umlet loses focus and shift is released afterwards

### ISSUE THREAD

**Issue by garretwilson** [umlet/umlet#446]
I'm using Eclipse 4.6.3 with Umlet 14.2. I place a "realizes" relationship between a class and an interface. I place the relationship line end circles on the edge of each entity. But then when I later move the class or the interface, the line stays put --- it doesn't seem to be attached.

I could swear this used to work. What's going on?

**Comment by garretwilson** [comment]
Yes, this is completely broken on Eclipse 4.6.3. Nothing is attached. I even pulled up an old class diagram to test it, and none of the objects are attached.

Even worse, I then tried to close the this class diagram I had loaded for testing, and it froze Eclipse --- the same bug from #310. It looks like I'm going to loose the changes I made to the original diagram.

This is so sad---it looks like Umlet is getting worse every day. Instead of bugs getting fixed, new ones appear.

The irony is that I'm writing a lesson on UML to teach to my students, and I would have recommended UMLet for them to use, but if I can't even successfully use UMLet myself when writing the lesson I don't know how I can recommend it to them.

This is all so sad---UMLet used to be a nice tool, except for some little bugs. Why is it that instead of fixing those bugs over the years, UMLet continues to collect more?

**Comment by garretwilson** [comment]
I tried this again from a fresh Eclipse restart, and the objects _are_ connected, as expected. But I've seen this problem before --- I think some glitch is causing it to appear. I'll try to look for patterns. (I was _not_ holding shift.)

**Comment by afdia** [comment]
I guess I have found the issue.
If the diagram loses the focus while you hold shift, then release shift and refocus the diagram, the key-release event is getting lost and the sticking behavior is still disabled.
The problem also exists in standalone UMLet (hold shift, click anywhere out of umlet, release shift, click into diagram)
I have fixed it for v14.3.

If it happens again in 14.2 try to press shift once and they should stick again.
